src/stan/io/dump.hpp

```diff
@@ -2,6 +2,9 @@
 #define STAN_IO_DUMP_HPP
 
 #include <stan/io/validate_zero_buf.hpp>
+#ifdef USE_STANC3
+#include <stan/io/validate_dims.hpp>
+#endif
 #include <stan/io/var_context.hpp>
```

Comment:
```suggestion
#ifdef USE_STANC3
#include <stan/io/validate_dims.hpp>
#endif

#include <stan/io/validate_zero_buf.hpp>
#include <stan/io/var_context.hpp>
```

Just trying to order the headers alphabetically and separate the `USE_STANC3` case.